src/_P016_IR.ino

```diff
@@ -417,12 +417,12 @@ boolean Plugin_016(byte function, struct EventStruct *event, String &string)
 
         if (nullptr != P016_data) {
           // convert result to uint32_t
-          uint32_t iCode = ((uint32_t) results.decode_type) * 0x1000000;  // Bits 31-24 (upper byte) for decode_type
+          uint32_t iCode = (static_cast<uint32_t> (results.decode_type)) * 0x1000000;  // Bits 31-24 (upper byte) for decode_type
           if (results.repeat)
             iCode += 0x800000;                                            // Bit 23 for repeat
           char strCode[P16_Cchars];
           if (safe_strncpy(strCode, resultToHexidecimal(&results), P16_Cchars)) {
-            iCode += strtol(strCode,0,16);                               // Bits 21-0 for code
+            iCode += strtoul(strCode,0,16);                               // Bits 21-0 for code
```

Comment:
We do have a `hexToUL` function.
And if the received HEX strings may exceed 8 HEX digits, then we might also need one `hexToULL` ???

Reply:
It seems like the protocol only supports 32 bit values. Tried to revive the msb that's missing from `strtol`.

Reply:
Yep, that msb part was clear, not that it doesn't allow for codes > 32 bit.

Reply:
At the end of the day, `hexToUL()` is calling `strtoul()`, but I've still changed it for readability & consistency.